profiler/src/main/java/com/splunk/opentelemetry/profiler/util/FileDeleter.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.profiler.util;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.function.Consumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public interface FileDeleter extends Consumer<Path> {
+
+  Logger logger = LoggerFactory.getLogger(FileDeleter.class);
+
+  static FileDeleter newDeleter() {
+    return path -> {
+      try {
+        Files.delete(path);
+      } catch (IOException e) {
+        logger.warn("Could not delete: " + path, e);
```

Comment:
```suggestion
        logger.warn("Could not delete: {}", path, e);
```

Reply:
Hmmm....I made this change, but I'm not confident that it will work.  The IDE says there are these overloads, none matching treats the exception as a Throwable, so I worry it would just error out trying to find a placeholder for that object?  Do you know if it works?
![image](https://user-images.githubusercontent.com/75337021/121050094-24eff380-c76d-11eb-97ca-abd1fc61f69a.png)

Reply:
IIRC, there's an overload with the throwable in the first slot to account for this case. But, I might be remembering some other library with a similar pattern.

Reply:
That would make sense...but I'm not seeing it here.

Reply:
It will work -- it's kinda non-obvious when you look at the signature, but slf4j checks the type of the last argument and if it's a `Throwable` then it adds the stack trace.

Reply:
Hah, ok cool then I'll leave it. I'm glad you knew that,  heh.

Reply:
I've checked it at least few times over the last few years 😂